hawkbit-ddi-resource/src/main/java/org/eclipse/hawkbit/ddi/rest/resource/DdiRootController.java

```diff
@@ -326,7 +344,10 @@ private ActionStatusCreate generateUpdateStatus(final DdiActionFeedback feedback
         }
 
         if (feedback.getStatus().getDetails() != null) {
-            messages.addAll(feedback.getStatus().getDetails());
+            messages.addAll(feedback.getStatus()
```

Comment:
Yes, I think we will remove the time stamp prefix. But will still just keep the message list flat without additional structure. We can leave it to controller to add meaning to the messages and reverse interpret them.

Other issue is that of getting the flattened list of messages directly from database using the filter. Having messages in `JpaActionStatus` as `ElementCollection` with a `CollectionTable` doesn't seem to work. What we are doing is actually a query similar to `.... detail_message NOT LIKE 'Update Server%'`. Criteria API cannot filter out the elements from the element collection, either all or none gets returned. The `CriteriaBuilder.notLike` expects `Element<String>` and not `Element<List<String>>`. This may be a JPA restriction so I'm not sure even if JPQL or Spring Query Methods is going to do that.

I'm considering following options:
1. Instead of restricting by count of messages, we restrict by count of action status, and then filter out the server generated messages in Java. We can use a method similar to `DeploymentManagement.findMessagesByActionStatusId` for this.
2. Change the mapping to OneToMany, but this required JPA model change which is something that I want to avoid.